File: CS/ChartsDemo.Wpf/ViewModels/CommentsData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("Comments")]
    public class CommentsData : List<CommentsInfo> {
        static List<CommentsInfo> dataSource = null;
        public static List<CommentsInfo> DataSource {
            get {
                if(dataSource == null) {
                    var s = new XmlSerializer(typeof(CommentsData));
                    dataSource = (List<CommentsInfo>)s.Deserialize(DataLoader.LoadFromResources("/Data/CommentsData.xml"));
                }
                return dataSource;
            }
        }
    }
    public class CommentsInfo {
        public string Category { get; set; }
        public List<CommentInfo> Items { get; set; }
    }
    public class CommentInfo {
        public DateTime Date { get; set; }
        public int CommentCount { get; set; }
    }
}

File: VB/ChartsDemo.Wpf/ViewModels/CommentsData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("Comments")>
    Public Class CommentsData
        Inherits List(Of CommentsInfo)


        Private Shared dataSource_Renamed As List(Of CommentsInfo) = Nothing
        Public Shared ReadOnly Property DataSource() As List(Of CommentsInfo)
            Get
                If dataSource_Renamed Is Nothing Then
                    Dim s = New XmlSerializer(GetType(CommentsData))
                    dataSource_Renamed = DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/CommentsData.xml")), List(Of CommentsInfo))
                End If
                Return dataSource_Renamed
            End Get
        End Property
    End Class
    Public Class CommentsInfo
        Public Property Category() As String
        Public Property Items() As List(Of CommentInfo)
    End Class
    Public Class CommentInfo
        Public Property [Date]() As Date
        Public Property CommentCount() As Integer
    End Class
End Namespace
